buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/classycle/Classycle.kt

```diff
@@ -40,30 +40,21 @@ import javax.inject.Inject
 
 
 @CacheableTask
-open class Classycle : DefaultTask() {
-
-    @get:Internal
-    lateinit var classesDirs: FileCollection
+open class Classycle @Inject constructor(
+    @get:Internal val classesDirs: FileCollection,
+    @get:Input val excludePatterns: Provider<List<String>>,
+    @get:Input val reportName: String,
+    @get:Internal val reportDir: File,
+    @get:InputFile @get:PathSensitive(PathSensitivity.NONE) val reportResourcesZip: RegularFileProperty,
+    private val antBuilder: IsolatedAntBuilder
```

Comment:
❌ Creating an `antBuilder` when the task is created seems to be too expensive - the [performance regression on the Gradle build itself](https://builds.gradle.org/repository/download/Gradle_Check_PerformanceTestCoordinator/12037842:id/report-performance-performance-tests.zip%21/report/tests/help-on-the-gradle-build-comparing-the-build.html) is real and probably caused by this.

We should either inject a factory or use method injection which is lazy.

Reply:
Went back to using method injection, since that was how it was handled before this PR.